gen/MotController_data.yml

```diff
@@ -61,8 +65,24 @@ c_MotController_GetDescription:
     raise NotImplementedError
 c_MotController_SetDemand:
   code: |
+    if self.hal_data['control_mode'] == ControlMode.Follower and mode != ControlMode.Follower:
+        follow_target = self.hal_data['follow_target']
+        hal_data['CAN'][follow_target]['followers'].remove(self._deviceId)
     self.hal_data['control_mode'] = mode
-    self.hal_data['value'] = demand0
+    if mode == ControlMode.PercentOutput:
+        self.hal_data['value'] = demand0 / 1023.
+    elif mode == ControlMode.Follower:
+        follow_target = demand0 & 0xFF
+        assert follow_target != self._deviceId
+        self.hal_data['follow_target'] = follow_target
+        hal_data['CAN'][follow_target]['followers'].append(self._deviceId)
+        self.hal_data['value'] = hal_data['CAN'][follow_target]['value']
+    else:
+        self.hal_data['value'] = demand0
+
+    for follower_id in self.hal_data['followers']:
+        hal_data['CAN'][follower_id]['value'] = self.hal_data['value']
+    return retval
```

Comment:
Don't return retval, it automatically gets returned for you (and ideally will get type checked too). I should probably add something to hooks.py to check for this...